src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java

```diff
@@ -307,5 +302,20 @@ public ResponseEntity<InputStreamResource> downloadCommonStats(
                 .body(isr);
     }
 
+    public CommonsPlus toCommonsPlus(Commons c) {
+        Optional<Integer> numCows = commonsRepository.getNumCows(c.getId());
+        Optional<Integer> numUsers = commonsRepository.getNumUsers(c.getId());
 
+        return CommonsPlus.builder()
+                .commons(c)
+                .totalCows(numCows.orElse(0))
+                .totalUsers(numUsers.orElse(0))
+                .build();
+    }
+
+    public Commons getEffectiveCapacity(Commons c) {
```

Comment:
```suggestion
    public Commons updateEffectiveCapacity(Commons c) {
```

This may seem picky, but I think you are setting a trap for future developers by calling this `getEffectiveCapacity`.   The problem is that there's a long established pattern than a method with the name `getFoo` is a method that:
* does NOT change the state of the object.
* returns the value of `foo`

But that's not what `getEffectiveCapacity` does. 

A better name might be `updateEffectiveCapacity`

What do you think?

Reply:
I agree, I will make this update.